services/checks/checks.go

```diff
@@ -226,24 +234,48 @@ func (s *Service) waitForResult(ctx context.Context, resultID string) ([]map[str
 			continue
 		}
 
-		// FIXME we still need to delete old result - they may never be done: https://jira.percona.com/browse/PMM-5840
 		if err = s.db.Delete(res); err != nil {
 			s.l.Warnf("Failed to delete action result %s: %s.", resultID, err)
 		}
 
 		if res.Error != "" {
-			return nil, errors.Errorf("Action %s failed: %s.", resultID, res.Error)
+			return nil, errors.Errorf("action %s failed: %s.", resultID, res.Error)
```

Comment:
A dot at the end of the `error` message should be added only to errors that are visible to users as-is – in our case, that's only (or mostly) gRPC statuses. In that case, we intentionally break the best practices because of the way we return gRPC statuses. (And even that decision is somewhat unfortunate; see https://jira.percona.com/browse/PMM-4695)

In other cases, `error` messages should be sentence fragments that are wrapped by the caller. That is the best practice in Go.

TL;DR: there is an extra dot in the end.